src/wallet/wallet_tx_builder.cpp

```diff
@@ -228,12 +228,16 @@ ResolveNetPayment(
     auto initialFee = fee.value_or(MINIMUM_FEE);
     return ValidateAmount(spendable, initialFee)
         .and_then([&](void) {
+            // Needed so that the initial call to `ResolvePayment` (which is just a placeholder used
+            // to help calculate the fee) doesn’t accidentally decrement funds.
+            auto tempMaxSapling = maxSaplingAvailable;
+            auto tempMaxOrchard = maxOrchardAvailable;
```

Comment:
```suggestion
            CAmount tempMaxSapling = maxSaplingAvailable;
            CAmount tempMaxOrchard = maxOrchardAvailable;
```
I guess `auto` *probably* won't infer a reference type? Still I think this would be a little clearer.

Reply:
`auto` will never infer a reference type; only `auto&` will do that.